### DISCUSSION TITLE
-morphooptiini error

### DISCUSSION THREAD

**Discussion by joshuajaistein** [neperfepx/neper#427] [Q&A] [answered]
neper -T -n 70 -morphooptiini "coo:file(neperinput.txt)" -tesrsize 5:4:3 -periodicity 'all' -format 'vtk'

The command ends up giving a ' You have discovered a bug in Neper!  Please file an issue at https://github.com/rquey/neper/issues.  Thank you.' error

I am trying to create geometry with my own coordinates in a 5:4:3 cube
[neperinput.txt](https://github.com/rquey/neper/files/8420163/neperinput.txt)

[neperinput.txt](https://github.com/rquey/neper/files/8420143/neperinput.txt)

**Comment by rquey** [comment, marked as answer]
First, use [-domain "cube(5,4,3)"](https://neper.info/doc/neper_t.html#cmdoption-domain) instead of `-tesrsize 5:4:3`.
Then, your seed file corresponds to a degenerate configuration where more than 4 cells intersect at a given vertex. Randomize the positions and it will be fine:
```
for i in 1 2 3
do
  gsl-randist $i 70 flat -1e-6 1e-6 > rand$i
done
paste neperinput.txt rand? | awk '{print $1+$4,$2+$5,$3+$6}' neperinput.txt > neperinput.txt2 
neper -T -n 70 -morphooptiini "coo:file(neperinput.txt2)" -domain "cube(5,4,3)" -periodicity 'all' -format 'tesr,vtk' -tesrsize 50:40:30
neper -V n70-id1.tesr -print img
```
![image](https://user-images.githubusercontent.com/2659376/161965107-4aeb6c83-e0c1-4ac2-a706-a727c23b60ce.png)
